### PR TITLE
Add logging helpers for `Dumpable` and `Yamlizable`

### PR DESCRIPTION
**Pull request by AlviseDeFaveri**

### CODE REVIEW

**Comment by AlviseDeFaveri** [comment]
Note: this PR has a twin MR in `revng-c`

File: include/revng/Support/IRHelpers.h [hunk 1]
```diff
@@ -984,6 +984,25 @@ inline std::string dumpToString(T &TheT) {
...
+  TheT.dump(std::declval<llvm::raw_ostream &>());
+};
+// clang-format on
+
+template<bool B, LLVMRawOStreamDumpable Dumpable>
+static void writeToLog(Logger<B> &L, const Dumpable &P, int /* Ignore */) {
+  if (L.isEnabled()) {
+    llvm::SmallString<32> Buffer;
+    llvm::raw_svector_ostream Stream(Buffer);
+    P.dump(Stream);
```
Comment (aleclearmind) [inline include/revng/Support/IRHelpers.h:1001]: I usually do like this in order to be sure that the stream is flushed, but it's probably not necessary.
```suggestion
    {
      llvm::raw_svector_ostream Stream(Buffer);
      P.dump(Stream);
    }
```

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: include/revng/Support/YAMLTraits.h [hunk 2]
```diff
@@ -218,3 +219,19 @@ llvm::Error serializeToFile(const T &ToWrite, const llvm::StringRef &Path) {
...
+    llvm::yaml::Output YAMLOutput(StringStream);
+    YAMLOutput << ToDump;
+  }
+  return Buffer;
+}
+
+template<Yamlizable T>
+std::string serializeToString(const T &Element) {
+  return serializeToString(const_cast<T &>(Element));
+}
```
[the remark this answers is not in the document]
Comment (aleclearmind) [reply include/revng/Support/YAMLTraits.h:237]: Your `serializeToString` could reuse the `serialize` function.
If you do so, the version taking `const T` can call the corresponding `serialize` function and we have one less `const_cast` in our codebase, which is good.

[the remark this answers is not in the document]
Comment (AlviseDeFaveri) [reply include/revng/Support/YAMLTraits.h]: keeping only the `const` version only and using `serialize()` in it appears to work smoothly, not sure it's the right way to do it though

**Comment by aleclearmind** [comment]
Merged in 16b7c61. Thanks!

**Pull request closed without merging**